integration-test-remote/src/test/java/io/cdap/cdap/app/etl/google/drive/GoogleDriveTest.java

```diff
@@ -390,7 +420,162 @@ public void testPartitionSize() throws Exception {
+      -30,-1,31,-99,29,108,-95,-52,-125,72,-44,-88,73,84,0,8,-85,65,106,83,-3,44,-5,-6,-6,7,-62,-105,32,
+      -23,115,33,-2,-49,0,0,0,0,73,69,78,68,-82,66,96,-126};
+    int contentLength = testPNGContent.length;
+
+    // create png file with metadata in Google Drive
+    // size: 174 bytes
+    createFile(service, testPNGContent, testFileName, "image/png", null, sourceFolderId);
+
+    Map<String, String> sourceProps = getDriveSourceMinimalDefaultConfigs();
+    sourceProps.put("fileTypesToPull", "binary,documents,spreadsheets");
+    sourceProps.put("bodyFormat", "bytes");
+    sourceProps.put("filter", String.format("name='%s'", testFileName));
+    sourceProps.put("fileMetadataProperties", "name,mimeType,size,imageMediaMetadata.width," +
+      "imageMediaMetadata.height,imageMediaMetadata.rotation");
+    sourceProps.put("maxPartitionSize", Integer.toString(testMaxPartitionSize));
+    Map<String, String> sinkProps = getFileSinkMinimalDefaultConfigs();
+
+    DeploymentDetails deploymentDetails =
+      deployApplication(sourceProps, sinkProps, "GoogleDriveSource", "FileSink",
+                        GOOGLE_DRIVE_PLUGIN_NAME, FILE_PLUGIN_NAME, GOOGLE_DRIVE_ARTIFACT, FILE_ARTIFACT,
+                        GOOGLE_DRIVE_PLUGIN_NAME + "-testWithFileSink");
+    startWorkFlow(deploymentDetails.getAppManager(), ProgramRunStatus.COMPLETED);
+
+    // check number of rows in and out
+    checkRowsNumber(deploymentDetails, 4);
+
+    Assert.assertTrue(Files.isDirectory(Paths.get(TMP_FOLDER)));
+
+    List<Path> allDeploysResults = Files.list(Paths.get(TMP_FOLDER)).collect(Collectors.toList());
+    assertEquals(1, allDeploysResults.size());
+
+    Path deployResult = allDeploysResults.get(0);
+    Assert.assertTrue(Files.isDirectory(deployResult));
+    assertEquals(1,
+                 Files.list(deployResult).filter(p -> p.getFileName().toString().equals("_SUCCESS")).count());
+
+    List<Path> destFiles =
+      Files.list(deployResult).filter(p -> p.getFileName().toString().startsWith("part")).collect(Collectors.toList());
+    assertEquals(4, destFiles.size());
+
+    JsonParser jsonParser = new JsonParser();
+
+    Map<Integer, byte[]> partitionedContent = new HashMap<>();
+    for (Path destFile : destFiles) {
+      List<String> fileLines = null;
+      try {
+        fileLines = Files.readAllLines(destFile);
+      } catch (IOException e) {
+        fail(String.format("Exception during reading file '%s': %s", destFile.toString(), e.getMessage()));
+      }
+      String fileContent = String.join(",", fileLines);
+      JsonElement rootElement = jsonParser.parse(fileContent);
+      Assert.assertTrue(rootElement.isJsonObject());
+
+      JsonObject rootObject = rootElement.getAsJsonObject();
+      Set<Map.Entry<String, JsonElement>> entries = rootObject.entrySet();
+
+      int resultOffset = 0;
+      byte[] resultBody = new byte[]{};
+      for (Map.Entry<String, JsonElement> entry : entries) {
```

Comment:
why do we need loop here? Can't we just do: rootObject.entrySet().get("name") rootObject.entrySet().get("mimeType") etc.
and than check number of keys.

Reply:
done